### PR TITLE
QA-Test tkt-53964 Adding API v2 tests for the new interfaces

### PR DESCRIPTION
**Pull request by ericbsd**

### CODE REVIEW

File: tests/api2/interfaces.py [hunk 1]
```diff
@@ -10,29 +10,134 @@
...
 
 
-@pytest.mark.skipif("BRIDGENETMASK" not in locals(),
-                    reason=BRIDGENETMASKReason)
-def test_03_get_interfaces_netmask():
+def test_03_get_interfaces_aliases_broadcast_ip():
     results = GET(f'/interfaces?name={interface}')
     assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
+    broadcast_ip = results.json()[0]['state']['aliases'][1]['broadcast']
```
Comment (william-gr) [inline tests/api2/interfaces.py]: Hard-coding index 1 is not quite right. You should loop through all of them.

File: tests/api2/interfaces.py [hunk 2]
```diff
@@ -10,29 +10,134 @@
...
+    assert broadcast_ip == broadcast, results.text
+
+
+def test_04_get_interfaces_aliases_type():
+    results = GET(f'/interfaces?name={interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
+    broadcast_ip = results.json()[0]['state']['aliases'][1]['type']
```
Comment (william-gr) [inline tests/api2/interfaces.py]: Same here, looping through all. broadcast_ip for that variable seems wrong.

File: tests/api2/interfaces.py [hunk 3]
```diff
@@ -10,29 +10,134 @@
...
+    aliases['type'] = broadcast_ip
+    assert isinstance(broadcast_ip, str) is True, results.text
+
+
+def test_05_get_interfaces_aliases_netmask():
+    results = GET(f'/interfaces?name={interface}')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list) is True, results.text
     netmask = results.json()[0]['state']['aliases'][1]['netmask']
```
Comment (william-gr) [inline tests/api2/interfaces.py]: Same here, index 1

File: tests/api2/interfaces.py [hunk 4]
```diff
@@ -10,29 +10,134 @@
...
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict) is True, results.text
+    assert results.json()['ipv4_dhcp'] is False, results.text
+
+
+def test_08_creating_vlan1_interface():
+    global payload
+    payload = {
+        "ipv4_dhcp": True,
+        "name": "vlan2",
```
Comment (william-gr) [inline tests/api2/interfaces.py]: vlan2 and tag 1 is unusual.
You better drop `name`, it should create "vlan1" automatically.

File: tests/api2/interfaces.py [hunk 5]
```diff
@@ -10,29 +10,134 @@
...
+def test_15_get_interfaces_checkin():
+    results = GET('/interfaces/checkin/')
+    assert results.status_code == 200, results.text
+    assert results.json() is None, results.text
+
+
+def test_16_get_interfaces_has_pending_changes():
+    results = GET('/interfaces/has_pending_changes')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), bool) is True, results.text
```
Comment (william-gr) [inline tests/api2/interfaces.py]: After /commit/ `has_pending_changes` should return False.

File: tests/api2/interfaces.py [hunk 6]
```diff
@@ -10,29 +10,134 @@
...
+
+
+def test_14_get_interfaces_checkin_waiting():
+    results = GET('/interfaces/checkin_waiting/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), bool) is True, results.text
+
+
+def test_15_get_interfaces_checkin():
+    results = GET('/interfaces/checkin/')
```
Comment (william-gr) [inline tests/api2/interfaces.py]: You should call /interfaces/commit/ to apply all the interfaces changes made before. You need to call /checkin/ after /commit/

File: tests/api2/interfaces.py (hunk 4 above)
Comment (ericbsd) [reply tests/api2/interfaces.py]: I did that to compare name with results output in test 09 and 11. and the name was valn1 have forgotten to put it back to vlan1.

**Comment by ericbsd** [comment]
thanks

**Pull request merged**